api/comparator.js

```diff
@@ -0,0 +1,72 @@
+/* jshint eqnull:true */
+var _ = require('underscore'),
+    Q = require('q');
+
+function parse(str) {
+    if (typeof str === 'string') {
+        try {
+            return JSON.parse(str, function(key, value) {
+                if (key === '__comment__') {
+                    return undefined;
+                }
+                return value;
+            });
+        } catch(e) {
+            console.warn(e);
+        }
+    }
+    return str;
+}
+
+var filters = [
+    function (data) { return data.replace(/([\.\[\(\$\?\+\*\\])/g, '\\$1'); },
+    function (data) { return data.replace(/&rx;(.*?)&rx;/g, function (match, param) {return param.replace(/\\(?!\\)/g, ''); }); }
+];
+
+function equals(obj1, obj2) {
```

Comment:
Your implementation of equals() is actually comparison. There are certain rules to follow, first off equals() should return boolean value (either is equal or not, we are not interested in know how much the object differ). 

Rules to obey when implementing equals()

* Reflexivity: x.equals(x) must be true
* Symmetry: x.equals(y) should be true iff y.equals(x) is true
* Transitivity: if x.equals(y) is true and y.equals(z) is true, then x.equals(z) is true
* Consistency: x.equals(y) must consistently return true or false as long as the state of x and y is not modified
* No object is equal to null; x.equals(null) is always false

See: http://jsclass.jcoglan.com/equality.html